src/lib/application-state-stores/explorer-stores.ts

```diff
@@ -0,0 +1,125 @@
+import { Readable, writable } from "svelte/store";
+import type { MetricsExplorerEntity } from "$lib/redux-store/explore/explore-slice";
+import type { MetricViewMetaResponse } from "$common/rill-developer-service/MetricViewActions";
+import type { TimeSeriesTimeRange } from "$common/database-service/DatabaseTimeSeriesActions";
+
+export interface MetricsExplorerStoreType {
+  entities: Record<string, MetricsExplorerEntity>;
+}
+const { update, subscribe } = writable({
+  entities: {},
+} as MetricsExplorerStoreType);
+
+const UpdateMetricsExplorer = (
+  id: string,
+  callback: (metricsExplorer: MetricsExplorerEntity) => void,
+  absenceCallback?: () => MetricsExplorerEntity
+) => {
+  update((state) => {
+    if (!state.entities[id]) {
+      if (absenceCallback) {
+        state.entities[id] = absenceCallback();
+      }
+      return state;
+    }
+    callback(state.entities[id]);
+    return state;
+  });
+};
+
+const Methods = {
+  sync(id: string, meta: MetricViewMetaResponse) {
```

Comment:
It looks like this `sync` logic has to do with selecting a `leaderboardMeasureId`. Shouldn't this logic go in the `LeaderboardMeasureSelector.svelte` component when that component fetches `/meta`, and we can remove this `sync` function?

If not, is there a better name than `sync`? I'm still a bit confused why it exists.

Reply:
If it has something to do with measures getting updated, then, when that happens, the `svelte-query` approach would be to invalidate the `/meta` query and refetch

Reply:
Hmmm we can revisit this in a later PR i feel. This is a quick and easy way to get the measures working.